src/router/navigationGuards.ts

```diff
@@ -25,36 +26,24 @@ export async function onLoginEnter(
                     ).data,
                 "Could not login."
             );
-            if (state.register) {
-                withErrorMessage(async () => {
-                    await axios.post(
-                        "/auth/api/login/registration/self-link",
-                        {
-                            register_token: tokenResponse.access_token
-                        },
-                        {
-                            headers: {
-                                Authorization: `Bearer ${await store.getAccessToken()}`
-                            }
-                        }
-                    );
-                }, "Could not link account.");
-            } else {
-                await handleOAuthResponse(tokenResponse, store);
-            }
+            await store.setNewTokenPair(tokenResponse.access_token, tokenResponse.refresh_token);
+
+            const redirectTo = store.redirectTo;
+            store.redirectTo = "";
             return {
-                path: state.from,
+                ...router.resolve(redirectTo),
                 replace: true
             };
         } catch (err) {
+            console.log(err);
```

Comment:
the frontend has an snackbar functionality to show error messages to the user (at least some information that sth went wrong)

Reply:
done